haptools/sim_genotype.py

```diff
@@ -93,20 +95,23 @@ def output_vcf(breakpoints, model_file, vcf_file, sampleinfo_file, out):
     # Iterate over VCF and output variants to file(in the beginning write out the header as well) until end of haplotype block for a sample (have to iterate over all samples each time to check)
     # Once VCF is complete we've output everything we wanted
     # VCF output have a FORMAT field where under format is GT:POP and our sample output is GT:POP ie 1|1:YRI|CEU
-    _write_vcf(breakpoints, hapblock_samples, vcf.samples, current_bkps, output_samples, vcf, out+".vcf")
+    _write_vcf(breakpoints, chroms, hapblock_samples, vcf.samples, current_bkps, output_samples, vcf, out+".vcf")
     return
 
-def _write_vcf(breakpoints, hapblock_samples, vcf_samples, current_bkps, out_samples, in_vcf, out_vcf):
+def _write_vcf(breakpoints, chroms, hapblock_samples, vcf_samples, current_bkps, out_samples, in_vcf, out_vcf):
     """
     in_vcf = cyvcf2 variants we are reading in
     out_vcf = output vcf file we output too
     """
     # output vcf file
     write_vcf = VariantFile(out_vcf, mode="w")
 
-    # make sure the header is properly structured
-    for contig in range(23):
-        write_vcf.header.contigs.add(str(contig+1))
+    # make sure the header is properly structured with contig names from ref VCF
+    for contig in in_vcf.seqnames:
+        # remove chr in front of seqname if present and compare
+        if re.search(r'X|\d+', contig).group() in chroms:
```

Comment:
will this still work if the contig isn't a canonical one? for example, what if the contig is named `chr1_KI270706v1_random`?